### PR TITLE
53 user must give birthday and gender when logging in

### PR DESCRIPTION
**Pull request by JoMuelle**
Created User now has correct Gender and Birthday assigned

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs22/constant/Gender.java [hunk 1]
```diff
@@ -1,5 +1,8 @@
 package ch.uzh.ifi.hase.soprafs22.constant;
 
 public enum Gender {
-    MALE, FEMALE, OTHER;
+    MALE,
+    FEMALE,
+    OTHER;
```
Comment (daveking7) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/constant/Gender.java:6]: see PR from Client

File: src/main/java/ch/uzh/ifi/hase/soprafs22/rest/dto/UserPostDTO.java [hunk 2]
```diff
@@ -31,4 +39,22 @@ public String getPassword() {
...
         this.password = password;
     }
+
+  public Date getBirthday () {return birthday;}
+
+  public void setBirthday(Date birthday) {this.birthday = birthday;}
+
+  public Gender getGender(){return this.gender; }
+
+  public void setGender(String gender) {
```
Comment (daveking7) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/rest/dto/UserPostDTO.java:49]: approved

**Review by daveking7 — CHANGES_REQUESTED** [review]
Good Solution, but make sure that you make a PR with a working solution, that is, first pull the Dev Branch locally in your branch, make sure it works, and then push it (First remove Merge Conflicts). And if you could update the tests too, that'd be great.

File: src/main/java/ch/uzh/ifi/hase/soprafs22/entity/User.java [hunk 3]
```diff
@@ -108,5 +108,22 @@ public void setStatus(UserStatus status) {
 
     public Gender getGender(){return this.gender; }
 
+    /**
+     * Method for Testing the gender of the user in JSON, as the gender should be
+     * as string and not ENUM there
+      * @return User.gender as string
+     */
+    public String getGenderString(){
```
Comment (daveking7) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/entity/User.java:116]: image
Wouldn't that make it much easier?

Comment (JoMuelle) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/entity/User.java:116]: Will make changes, forgot you could do that

Comment (JoMuelle) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/entity/User.java:116]: If done tests stop passing, not sure why, propose to leave it as it is?

Comment (daveking7) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/entity/User.java:116]: In UserGetDTO, change getGender return type back to Gender, then it should work (atleast for the tests)

Comment (JoMuelle) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/entity/User.java:116]: Ahh nice, thanks, works now.

**Pull request merged**